[18.0][IMP] spreadsheet_oca: Add Kanban views

flotho opened the pull request:
Also add active option

<img width="1183" height="370" alt="image" src="https://github.com/user-attachments/assets/21633667-c780-4ec7-8964-8be93bc0dca7" />

flotho committed:
[IMP] spreadsheet_oca:
* Add Kanban views
* Also add active option

[IMP] spreadsheet_oca:
* Simplify partner usage
* Add tags
* Improve kanban view

[IMP] spreadsheet_oca: Improve form view

[IMP] spreadhseet_oca: add tag menu

[IMP] spreadhseet_oca: add default image

IMP: Add tag on view

[REF] spreadsheet_oca: no more partner

[REF] spreadsheet_oca: no more partner

[REF] spreadsheet_oca: no more partner

[REF]rename file
[ACL]Ad specific rule for manager

flotho commented:
ping @DorianMAG @JulienMartinez

legalsylvain commented:
Hi. 
- adding active field : :heavy_check_mark: 
- adding kanban view : :heavy_check_mark: 
- adding partner / comercial partner field : I dont' see the interest, and I don't see it very clear. Could you elaborate ? 
thanks !

flotho commented:
Hi @legalsylvain 
Thanks for the code review
>     * adding partner / comercial partner field : I dont' see the interest, and I don't see it very clear. Could you elaborate ?
>       thanks !

Most of time, we (Mind And Go / Datacup) are using sheets for a specific project, customer etc. We use sheets for computation but also for sharing simple quality process between us. Basically some checklist with colors.
Having a partner on sheets really helps to give context and find easily the document and prevent us to add partner name in the file name.

legalsylvain commented:
Hi @flotho. Thanks for your reply ! 

- Could we imagine having a M2M fields regarding partners. it could be more generic. (and image can be computed as the image of the first partner).
- I see the "commercial" partner quite specific. I mean, for the time being, this field is present only in sale / account module. for all the other modules. (crm / project / etc...) there is a simple partner_id.


@etobella : a point of view here ? 

thanks !

etobella commented:
you are solving the issue by using partner, but wouldn't be cleaner to use res_model and res_id and access/generate them from the thread similar to mgmtsystem.nonconformity?

flotho commented:
Hum, this leads us to a great path.
we're using commercial partner_id not for commercial purpose but to be able to group by customer company and not by the contact.
I'm totally fan of the M2M field @legalsylvain 
@etobella , I'll have a look to your proposal.
Anyway, to ease the merging what could be acceptable in the first step ?
M2m on partner adn without commercial_partner_id ?
Regards

legalsylvain commented:
>  Anyway, to ease the merging what could be acceptable in the first step ?

I let @etobella and other main developpers answer on that topic. I didn't contributed on that great repo, and so don't have a clear PoV.

chrisandrewmann commented:
Thanks @flotho it's great to see and i'm 100% behind a kanban view.
My thoughts below, see the attached mockup image:

- I think the icon should be static and possibly changeable - linked to an image field on the spreadsheet.
Alternatively, a mechanism to generate a thumbnail preview could be used similar to Enterprise?
- Cursor change and clickable image on left
- Menu with 3 options (Open, Settings, Delete)
- I'd support having the ability to set a partner_id, but not the commercial partner as I think that's possibly confusing
- Instead of defaulting to commercial partner for grouping i'd suggest adding M2M tags, which could be displayed on the kanban card and also easily searchable / groupable

<img width="1140" height="649" alt="image" src="https://github.com/user-attachments/assets/7230dd0e-8791-4ec3-886d-1f263c777c33" />

flotho commented:
Thanks,
Here is my new proposal :
<img width="472" height="261" alt="image" src="https://github.com/user-attachments/assets/5e64e700-edca-4bfc-93f8-f02d19805798" />

legalsylvain commented:
Hi @flotho. I put the PR in draft state, as it looks WIP, with red CI. Feel free to mark it as "Ready for review", once all is green and commit have been refactored.
thanks for this improvement !

chrisandrewmann commented:
@flotho looks really good, I like the new proposal.

Couple of things:
- Should add a Configuration > Tags menu to manage the tags
- If the image is missing, there should be a default placeholder (suggested the font-awesome icon of a spreadsheet in my example) - Otherwise it's empty whitespace

flotho commented:
> @flotho looks really good, I like the new proposal.
> 
> Couple of things:
> 
>     * Should add a Configuration > Tags menu to manage the tags
> 
>     * If the image is missing, there should be a default placeholder (suggested the font-awesome icon of a spreadsheet in my example)

you're right.
Yet I won't do it no.
Is it blocking for merging ?

chrisandrewmann commented:
> > @flotho looks really good, I like the new proposal.
> > Couple of things:
> > ```
> > * Should add a Configuration > Tags menu to manage the tags
> > 
> > * If the image is missing, there should be a default placeholder (suggested the font-awesome icon of a spreadsheet in my example)
> > ```
> 
> you're right. Yet I won't do it no. Is it blocking for merging ?

I'll wait for other comments @legalsylvain @etobella ?
Personally I think we should fix these 2 points for usability.

<img width="1023" height="371" alt="image" src="https://github.com/user-attachments/assets/6395be4e-61f6-4432-a69b-aebdddfd00b6" />

flotho commented:
ping @legalsylvain @etobella @chrisandrewmann , ready for review

chrisandrewmann commented:
@flotho thanks, changes look good to me.
Quickly tested on runbot.
However CI tests failed due to missing alt tag on img. Should be simple fix, then we can review

flotho commented:
> @flotho thanks, changes look good to me. Quickly tested on runbot. However CI tests failed due to missing alt tag on img. Should be simple fix, then we can review

done !

chrisandrewmann reviewed (approved):
Looks good to me, approved. Will wait for others

flotho commented:
Hi @etobella @CarlosRoca13 @chrisandrewmann 
Any chance to have a review on this small addition ?
Regards

etobella reviewed (commented):
I am in favor of the concept of kanban and tags, but I think it will be better to make the res_model, res_id strategy.

flotho commented:
> I am in favor of the concept of kanban and tags, but I think it will be better to make the res_model, res_id strategy.

Noted, thanks for the feedback

pedrobaeza commented:
@flotho if you remove the `partner_id` part of this PR, this can be merged, and you can put that part in a extension on your side, or propose the generic way. You can also use a reference field.

flotho commented:
Thanks @pedrobaeza , it's done ,
ping @etobella your review is welcomed

pedrobaeza commented on spreadsheet_oca/models/__init__.py:
```diff
@@ -1,4 +1,5 @@
 from . import spreadsheet_abstract
+from . import spreadsheet_spreadsheet_tags
```
The file name should be singular.

pedrobaeza commented on spreadsheet_oca/security/ir.model.access.csv:
```diff
@@ -4,3 +4,4 @@ access_spreadsheet_oca_revision,access_spreadsheet_oca_revision,model_spreadshee
 spreadsheet_oca.access_spreadsheet_spreadsheet_import,access_spreadsheet_spreadsheet_import,spreadsheet_oca.model_spreadsheet_spreadsheet_import,base.group_user,1,1,1,1
 access_spreadsheet_import_mode,access_spreadsheet_oca_revision,model_spreadsheet_spreadsheet_import_mode,base.group_user,1,0,0,0
 access_spreadsheet_select_row_number,access_spreadsheet_select_row_number,model_spreadsheet_select_row_number,base.group_user,1,1,1,1
+access_spreadsheet_spreadsheet_tag,access_spreadsheet_spreadsheet_tag,model_spreadsheet_spreadsheet_tag,group_user,1,1,1,1
```
Only spreadsheet managers should change tags I think.

legalsylvain replied:
That's a matter of taste, IMO.

```
./src/OCA/manufacture/mrp_bom_tag/security/ir.model.access.csv:access_mrp_bom_tag_manager,Bill Of Material Tag Manager,model_mrp_bom_tag,mrp.group_mrp_user,1,1,1,1
./src/OCA/server-auth/vault/security/ir.model.access.csv:access_vault_tag,access_vault_tag,model_vault_tag,base.group_user,1,1,1,1
./src/odoo/addons/website_forum/security/ir.model.access.csv:access_forum_tag_user,forum.tag.user,model_forum_tag,base.group_user,1,1,1,1
./src/odoo/addons/repair/security/ir.model.access.csv:access_repair_tag_user,Repair Tags user,model_repair_tags,stock.group_stock_user,1,1,1,1

```

etc.

pedrobaeza replied:
The examples you are putting, except the first one, it's because there's no manager role. Usually, all the "master data" are only accessible by managers of the area.

legalsylvain replied:
> The examples you are putting, except the first one, it's because there's no manager role. 

I don't get it. 
Regarding `model_repair_tags` for exemple. It is possible to manage tag with `stock.group_stock_user` group, not only `stock.group_stock_manager`.   (group_stock_manager implies group_stock_user)
Or did I missed something ?

pedrobaeza replied:
Please don't go off-topic. The topic here is that it's legit to not allow to modify tags to "simple" spreadsheet users, just the managers. That's it.

legalsylvain replied:
Is it possible for you to speak correctly and on valid points?
That's not the first time, you're acting like an authoritarian leader. It's annoying to talk with you.
Thank you for questionning yourself on this point.

pedrobaeza replied:
It's you the one going to off-topics, as I'm talking about spreadsheet tags, not repair tags. And I'm being totally correct. Please tell me in which moment I have not been correct. It's also annoying to talk to you, deviating conversations that are not targeted to you, but to @flotho, who is the author of the PR.

pedrobaeza reviewed (commented):
Please reduce commits into 2 (tags and kanban)

pedrobaeza commented:
Thanks for the renaming, @flotho 

Can you please squash commits as suggested, and what do you think about the tag permissions?

flotho commented:
done @pedrobaeza

pedrobaeza commented:
@flotho I was talking about 2 commits, as you are doing here 2 things: tags and kanban view. The commit message also lacks the first line explanations. And the tag question is still open.

flotho commented:
oh damned I squashed too much ;-)
Do I need to undo ?
regarding ACL I've already changed the rules to have 2 one for basic user and another for manager

pedrobaeza commented:
Thank you for the change in the tags. About the commits, if it's not very difficult to you, it would be great to have both features split, but I don't want to press more. Just tell me and let's finish this.

chrisandrewmann commented on spreadsheet_oca/views/spreadsheet_spreadsheet.xml:
```diff
@@ -42,17 +61,30 @@
-                    <h1>
-                        <field name="name" placeholder="Name" />
-                    </h1>
+                    <field name="badge_image" widget="image" class="oe_avatar" />
```
To enable "Archived" ribbon and ability to archive from form, add above:

```
<field name="active" invisible="1"/>
<widget name="web_ribbon" title="Archived" bg_color="text-bg-danger" invisible="active"/>
```

chrisandrewmann commented on spreadsheet_oca/views/spreadsheet_spreadsheet.xml:
```diff
@@ -78,12 +110,101 @@
         </field>
```
Noticed you added inherited "mail.thread", "mail.activity.mixin" to spreadsheet_spreadsheet.py but didn't enable chatter in the view. Think this would be a welcome simple change. Just add this line after the ending sheet tag:

`<chatter/>`

legalsylvain commented on spreadsheet_oca/security/ir.model.access.csv:
```diff
@@ -4,3 +4,5 @@ access_spreadsheet_oca_revision,access_spreadsheet_oca_revision,model_spreadshee
 spreadsheet_oca.access_spreadsheet_spreadsheet_import,access_spreadsheet_spreadsheet_import,spreadsheet_oca.model_spreadsheet_spreadsheet_import,base.group_user,1,1,1,1
 access_spreadsheet_import_mode,access_spreadsheet_oca_revision,model_spreadsheet_spreadsheet_import_mode,base.group_user,1,0,0,0
 access_spreadsheet_select_row_number,access_spreadsheet_select_row_number,model_spreadsheet_select_row_number,base.group_user,1,1,1,1
+access_spreadsheet_spreadsheet_tag,access_spreadsheet_spreadsheet_tag,model_spreadsheet_spreadsheet_tag,spreadsheet_oca.group_user,1,0,1,0
```
```suggestion
access_spreadsheet_spreadsheet_tag,access_spreadsheet_spreadsheet_tag,model_spreadsheet_spreadsheet_tag,spreadsheet_oca.group_user,1,1,1,0
```

At least. Otherwise, if a user create a tag, ans if he made a mistake in the name, he can not change it.

pedrobaeza replied:
Well, I still think that users should only use them (read them), not to modify them. If they are going to be able to create them, but not to remove them, that's worst.

legalsylvain replied:
What is your opinion here @flotho, and what do you need ?

flotho replied:
Thanks for sharing your PoV.
I'm personally used to install the module that prevent creation on the fly for a user because in the end it produce inconsistent data. Typically the use case @legalsylvain described : `if a user create with a bad name`
That's why I'm in favor of @pedrobaeza to prevent user to modify the tag. Keeping data consistent should be the responsibility of `managers`.

pedrobaeza replied:
¿Then should it be this?:

```suggestion
access_spreadsheet_spreadsheet_tag,access_spreadsheet_spreadsheet_tag,model_spreadsheet_spreadsheet_tag,spreadsheet_oca.group_user,1,0,0,0
```

legalsylvain reviewed (commented):
A minor remark. Otherwise, LGTM. 

Thanks for this improvment.

flotho committed:
[IMP]Active ribbon and chatter on view

chrisandrewmann reviewed (approved):
Amazing, thanks for that. Looks good.

flotho committed:
[IMP]ACL: make tag readonly for basic users

pedrobaeza reviewed (approved):
Let's merge this although there's an extra commit for the last change.

/ocabot merge minor
